### PR TITLE
Set gradient flag in forward benchmarks

### PR DESCRIPTION
**Pull request by Priya2698**
Set `requires_grad=True` flag for forward benchmarks.

### CODE REVIEW

**Comment by Priya2698** [comment]
!test --pybench --dev

**Comment by Priya2698** [comment]
!test --pybench --dev

**Comment by Priya2698** [comment]
!test --pybench --dev

**Comment by Priya2698** [comment]
I'll take another pass over the benchmarks, and check for this. If you notice any other benchmark requiring this, please let me know.

**Review by jjsjann123 — APPROVED** [review]
Thanks a lot for the clean up.

It always felt a bit ambiguous to me on when/whether we should be running with grad on the forward only benchmark.

But I think I agree with your judgement here. i.e. if it's an inference only benchmark, the test name should probably include some keyword indicating that. And if there's a `fwd` already in the name we probably want to keep the bwd graph feasible.

**Review by liqiangxl — APPROVED** [review]
Can you check performance change?

**Comment by Priya2698** [comment]
> It always felt a bit ambiguous to me on when/whether we should be running with grad on the forward only benchmark.

I see your point. we can be measuring either performance. One caveat here is, that our IOBytes computation is static, and is estimated based on training time runs. I am looking into how to automate that, but haven't found a robust method yet. That should remove these inconsistencies.

Edit: Attempting this at https://github.com/NVIDIA/Fuser/tree/pm/iobytes. Will compare and update the group on if this approach works.

**Comment by Priya2698** [comment]
!test --pybench-full --dev

**Comment by Priya2698** [comment]
Benchmark results at: http://nv/eM/

 I have summarized the benchmarks that were modified in this PR.
 The embedding benchmarks are not loading for me, I'll try adding them to the table later.

Values are geomean of speedup ratio (inverse execution time): execution_time (weekly) / execution_time (nvfuser) 

| Benchmark                    | Thunder-nvfuser | Torchcompile | Eager |
|------------------------------|-----------------|--------------|-------|
| Dropout_layernorm_fwd        | 0.93            | 1.06         | 1.0   |
| Dropout_rmsnorm_bwd          | 0.78            | 0.73         | 0.86  |
| Dropout_rmsnorm_fwd          | 0.92            | 1.23         | 1.0   |
| huggingface_attn_fwd         | 0.74            | 0.92         | 1.0   |
| nanogpt_attn_fwd             | 0.70            | 0.84         | 1.0   |
| layernorm_fwd                | 1.0             | 1.0          | 1.0   |
| rmsnorm_fwd                  | 1.0             | 1.0          | 1.0   |
| silu_mul_fwd                 | 1.0             | 1.0          | 1.0   |
| Softmax_fwd (reduction axis: 0/1) | 1.0        | 1.0          | 1.0   |
